### ISSUE TITLE
Want to delete tags on Docker Hub? Good luck!

### ISSUE THREAD

**Issue by crohr** [crohr/ebarnouflant#5]
*tl;dr* Docker is all hype these days, but some parts (like the Docker Hub) are definitely lacking some love.

When I use a tool such as [Empire](https://github.com/remind101/empire) for deploying docker containers, I like to tag and push each commit of the corresponding repo on Docker Hub. This allows to easily pin or rollback to a specific release with `emp deploy my-org/my-app:SHA`.

However the performance of the Hub decreases a lot whenever you pass the 100 tags mark, like when you try to fetch the details of a tag via the (undocumented?) Docker Hub API:

```
curl -u user:pass https://index.docker.io/v1/repositories/my-org/my-app/tags/latest
# => 504 Gateway Timeout
```

First you need to open a support request, then you get asked the list of tags you want to keep, then you wait for a few days hoping that your request will finally go through, and then you get this message:

> Our scripts only take in a list of tags to be deleted, instead of the other way round. Can you send us the full list of tags to be deleted?

Which you can't reply to, since there is no way currently to programmatically fetch all your tags, and the Hub UI only displays the last 100 or so.

I guess your best bet is then to destroy the Hub repo, re-create it, and push back all the tags you need...

Or maybe **you** know a better solution? Please let me know in the comments.
